exceptions/exceptions.hh

```diff
@@ -67,6 +67,7 @@ enum class exception_code : int32_t {
     FUNCTION_FAILURE= 0x1400,
     WRITE_FAILURE   = 0x1500,
     CDC_WRITE_FAILURE = 0x1600,
+    CDC_PREIMAGE_READ_FAILURE = 0x1601,
```

Comment:
Are we allowed to allocate new exception codes like that? This error code is not recognized by Cassandra. It is dangerously close to CDC_WRITE_FAILURE, and I can imagine a possibility where new CDC-related codes could be defined right after CDC_WRITE_FAILURE (`0x1601`, `0x1602` etc.).

Avi suggested to allocate error codes starting from `0x1aff` backwards, see discussion here: https://github.com/scylladb/scylla/pull/5537#discussion_r363084064

Reply:
OK, I picked `0x1aef` for `CDC_PREIMAGE_READ_FAILURE`, keeping 15 free slots for more overflow-related codes.
Python driver seems to handle non-standard codes nicely, but I'll get an opinion from someone with more "strategic" knowledge.

Reply:
It turns out that some drivers convert unknown error numbers into protocol errors, so we can't do this. You can use CDC_WRITE_FAILURE instead.